mouseion/mouseion/train_and_build_index.py

```diff
@@ -4,41 +4,42 @@
 import tqdm
 import pathlib
 
-from .indexbuilder import GPUIndexBuilder, IndexBuilder
+from indexbuilder import GPUIndexBuilder, IndexBuilder
 
 def main(args):
     """
     Determine if an output directory exists.
     """
-    if not pathlib.Path(args.output).is_dir():
-        raise ValueError(f"The output path is invalid {args.output}")
 
     """
     Combines numpy embeddings from multiple npy files into a FAISS index.
     """
     try:
+        print(f"preload")
         spec = json.load(args.spec.open("r"))
+        print(f"spec={spec}")
     except Exception as e:
         print(f"Error reading spec file: {e}")
         return
 
-    total_embeddings = sum([numpy.load(shard["embeddings"], mmap_mode="r").shape[0] for shard in spec["shards"]])
+    total_embeddings = sum([numpy.load(shard["embeddings"], mmap_mode="r").shape[0] for shard in spec])
+    print(f"total embeddings={total_embeddings}")
     is_trained = False
 
     try:
-        first_shard = numpy.load(spec["shards"][0]["embeddings"], mmap_mode="r")
+        first_shard = numpy.load(spec[0]["embeddings"], mmap_mode="r")
     except Exception as e:
         print(f"Error loading shard: {e}")
         return
 
     dim = first_shard.shape[1]
     print(f'dimensionality is {dim}')
 
-    builder_cls = GPUIndexBuilder if args.use_gpus else IndexBuilder
+    builder_cls = GPUIndexBuilder
```

Comment:
Is the `IndexBuilder` import no longer required? Not worried about structure, just trying to understand whether this was a typo.

Reply:
I am not sure why I made this change. Looking at the change though, it appears to disable cpu index building. I think I may have done this because of problems building faiss. I have sorted out the faiss problems with the latest fiass 1.8.0 build, so maybe this isn't needed. I will reset this PR to remove these files.